src/least_cost_path/least_cost_path.py

```
@@ -285,129 +316,516 @@
+            lcp_layer, temp_file_path  # Используем временный файл вместо пути в project_folder
+        )
+
+        # Добавляем слой в проект (если нужно)
+        QgsProject.instance().addMapLayer(watershed_layer)
+
+    # Очистка временных файлов
+        for temp_file in [dem_reprojected, dem_pooled]:
+            try:
+                os.remove(temp_file)
+            except:
+                pass
```

Ruff (0.11.9)

Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(Exception)`

(SIM105)